system/db/content.go

```diff
@@ -36,6 +36,98 @@ func SetContent(target string, data url.Values) (int, error) {
 	return update(ns, id, data)
 }
 
+// UpdateContent inserts or updates values in the database.
+// Updated content will be merged into existing values from the database.
+// The `target` argument is a string made up of namespace:id (string:int)
+func UpdateContent(target string, data url.Values) (int, error) {
+	t := strings.Split(target, ":")
+	ns, id := t[0], t[1]
+
+	// check if content id == -1 (indicating new post).
+	// if so, run an insert which will assign the next auto incremented int.
+	// this is done because boltdb begins its bucket auto increment value at 0,
+	// which is the zero-value of an int in the Item struct field for ID.
+	// this is a problem when the original first post (with auto ID = 0) gets
+	// overwritten by any new post, originally having no ID, defauting to 0.
+	if id == "-1" {
```

Comment:
Assuming then that we consolidate to the new SetContent, we'll want some level of id validation in the handler also, just so even if the content is Externalable/Addable its getting created through the correct handler with the appropriate hooks for create vs. update.